File: Arrays/EX1/Book.java
public class Book {

    private int ISBN;
    private String title;
    private String publisher;
    private int price;

    public Book() {
        this.ISBN = 0;
        this.title = "";
        this.publisher = "";
        this.price = 0;
    }

    public Book(int ISBN, String title, String publisher, int price) {
        this.ISBN = ISBN;
        this.title = title;
        this.publisher = publisher;
        this.price = price;
    }

    public void setISBN(int iSBN) {
        ISBN = iSBN;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setPublisher(String publisher) {
        this.publisher = publisher;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public int getISBN() {
        return ISBN;
    }

    public String getTitle() {
        return title;
    }

    public String getPublisher() {
        return publisher;
    }

    public int getPrice() {
        return price;
    }

    public String toString() {
        return ("ISBN: " + getISBN() + "\nTitle:" + getTitle() + "\nPublisher:" + getPublisher() + "\nPrice: "
                + getPrice());
    }

}

File: Assignments/as5/book.cpp
#include "book.h"

Book::Book() : Medium(), publisher("")
{
}

Book::Book(const std::string abbreviation, const std::string author,
           const std::string title, const unsigned int yearOfPublication,
           std::string publisher) : Medium(abbreviation, author, title, yearOfPublication), publisher{publisher}
{
}

Book::~Book() {}

void Book::setPublisher(std::string publisher) { this->publisher = publisher; }
std::string Book::getPublisher() { return this->publisher; }
std::string Book::toString() const
{
    return (Medium::toString() + '\n' + this->publisher + ',' + std::to_string(Medium::getPublicationYear()));
}
